### ISSUE TITLE
Derived->Default dependency is broken

### ISSUE THREAD

**Issue by almozavr** [immutables/immutables#196]
During `MyItem.from(anotherMyItem)` `@Derived` attr got wrong `@Default` value: seems like it calls default impl. while it's (non-default) value haven't been copied yet. 
```java
@Value.Immutable
abstract class MyItem {
    @Value.Derived public String index() {
        return uuid() + "!!!";
    }

    @Value.Default public String uuid() {
        return UUID.randomUUID().toString();
    }

    String title();
}
...
MyItem source = ImmutableMyItem.builder().title("1").build(); // uuid is 5 e.g., index = 5!!!
MyItem replica =  ImmutableMyItem.copyOf(source).withTitle("2"); // replica's uuid = 5, index = *randomValue*!!!
...
```
Problem resides in constructor which is called from `withTitle()` - there is a call to `super.index()` and to `uuid()` as well, but `uuid()` uses `initShim` by that moment which is `not null`, so wrong value is applied.

**Comment by almozavr** [comment]
I think the problem is `initShim`'s `uuid` is not initialized in constructor even `super.index` will use it as `initShim` is not null by default.

**Comment by almozavr** [comment]
Possible solution:
```java
private ImmutableMyItem(String title, String uuid) {
    this.title = ImmutableMyItem.requireNonNull(title);
    if (uuid != null) {
      initShim.uuid(uuid);
    }
    this.uuid = initShim.uuid();
    this.index = ImmutableMyItem.requireNonNull(super.index());
    this.initShim = null;
}
```

**Comment by elucash** [comment]
That's right, fixing broken copy and non-builder constructors.
In a meantime while it's not released yet, you may switch to previous mode (prior to 2.1.0) by using `Value.Style(unsafeDefaultAndDerived = true)` style.
Thank you for catching this bug!

**Comment by elucash** [comment]
Fix released in 2.1.1. Thank you!
